tests/integration/conftest.py

```diff
@@ -421,70 +413,46 @@ def run_job_cmd(
     return _disable_polling
 
 
-@pytest.fixture
-def _setup_validate_cli():
-    """Provides CLI arguments for cylc validate."""
-    def _inner(opts=None):
-        setup = SimpleNamespace()
-        setup.parser = validate_gop()
-        setup.opts = setup.parser.get_default_values()
-        setup.opts.templatevars = []
-        setup.opts.templatevars_file = []
-        if setup.opts is not None:
-            for key, val in opts.items():
-                setattr(setup.opts, key, val)
-        return setup
+@pytest.fixture(scope='module')
+def mod_workflow_source(mod_flow, mod_one_conf, tmp_path_factory):
+    """Create a workflow source directory.
+
+    Args:
+        cfg: Can be passed a config dictionary.
+
+    Yields:
+        Path to source directory.
+    """
+    def _inner(cfg=None):
+        src_dir = _make_src_flow(tmp_path_factory.getbasetemp(), cfg)
+        return src_dir
     yield _inner
 
 
 @pytest.fixture
-def source_workflow(run_dir, flow_src):
-    """Create a workflow in source directory and install it.
-
-    Useful for testing interaction between installed and source directories.
+def workflow_source(mod_flow, mod_one_conf, tmp_path):
+    """Create a workflow source directory.
 
     Args:
         cfg: Can be passed a config dictionary.
 
     Yields:
-        A namespace object with properties:
-            src: source location.
-            opts: options set for cylc install.
-
+        Path to source directory.
     """
-    source = flow_src({
-        'scheduler': {
-            'allow implicit tasks': True
-        },
-        'scheduling': {
-            'initial cycle point': '1854',
-            'graph': {
-                'R1': 'foo'
-            }
-        }
-    })
-
-    workflow_name = 'cit-' + str(source.name)
-
     def _inner(cfg=None):
-        # Object to store info about the setup.
-        setup = SimpleNamespace()
-
-        # Create a workflow source:
-        if cfg is None:
-            setup.src = source
-        else:
-            setup.src = flow_src(cfg)
-
-        # Setup Opts:
-        setup.opts = install_gop().get_default_values()
-        setup.opts.no_run_name = True
-        setup.opts.workflow_name = workflow_name
+        src_dir = _make_src_flow(tmp_path, cfg)
+        return src_dir
+    yield _inner
 
-        # Carry out the installation:
-        install(setup.opts, setup.src)
-        setup.flow_file = setup.src / 'flow.cylc'
-        return setup
 
+@pytest.fixture
+def cylc_install(test_dir, run_dir):
+    """Install a workflow from source"""
+    def _inner(source, **kwargs):
+        Opts = Options(install_gop())
```

Comment:
This will create `Opts` every time that `_inner` is called, which could be a lot.

Should create a constant to hold this value so it doesn't have to be re-computed e.g:

```python
InstallOptions = Options(install_gop())

...

@pytest.fixture
def cylc_install(test_dir, run_dir):
    ...
```